### PR TITLE
core[patch]: propagate `parse_docstring` to tool decorator

### PR DESCRIPTION
**Pull request by ccurme**
Disabled by default.

```python
from langchain_core.tools import tool

@tool(parse_docstring=True)
def foo(bar: str, baz: int) -> str:
    """The foo.

    Args:
        bar: this is the bar
        baz: this is the baz
    """
    return bar


foo.args_schema.schema()
```
```json
{
  "title": "fooSchema",
  "description": "The foo.",
  "type": "object",
  "properties": {
    "bar": {
      "title": "Bar",
      "description": "this is the bar",
      "type": "string"
    },
    "baz": {
      "title": "Baz",
      "description": "this is the baz",
      "type": "integer"
    }
  },
  "required": [
    "bar",
    "baz"
  ]
}
```

### CODE REVIEW

File: libs/core/tests/unit_tests/test_tools.py [hunk 1]
```diff
@@ -959,6 +959,19 @@ def foo(bar: str, baz: int) -> str:
         "required": ["bar", "baz"],
     }
 
+    foo2 = tool(foo, parse_docstring=True)
```
Comment (eyurtsev) [inline libs/core/tests/unit_tests/test_tools.py:962]: Could you add a test that shows that this throws an error it the doc-string does not follow the correct format?

**Comment by eyurtsev** [comment]
1. Could you define the behavior to raise an exception on invalid doc-strings (shown below)
2. Could you add a test that verifies what happens if the tool requests the `run_manager`?


```python
from langchain_core.tools import tool

@tool(parse_docstring=True)
def foo(bar: str, baz: int) -> str:
    """The foo.

    Args:
        banana: this is the bar
        monkey: this is the baz
    """
    return bar


foo.args_schema.schema()
```

----

**Comment by ccurme** [comment]
> 2. Could you add a test that verifies what happens if the tool requests the `run_manager`?

Can you clarify more specifics here? There is an existing test but I'm not sure if it's missing a condition:
https://github.com/langchain-ai/langchain/blob/16e178a8c2c26c6f7d14825ea7c0d0a4a91b8a59/libs/core/tests/unit_tests/test_tools.py#L431-L446

File: libs/core/tests/unit_tests/test_tools.py (hunk 1 above)
Comment (ccurme) [reply libs/core/tests/unit_tests/test_tools.py:963]: done

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**